pkg/tests/tasks/traffic/egress/access_external_services_test.go

```diff
@@ -41,32 +42,29 @@ func TestAccessExternalServices(t *testing.T) {
 
 		t.Log("This test validates accesses to external services")
 
-		assertRequestSuccess := func(url string) {
-			execInSleepPod(t, ns, buildGetRequestCmd(url),
-				assert.OutputContains("200",
-					fmt.Sprintf("Got expected 200 OK from %s", url),
-					fmt.Sprintf("Expect 200 OK from %s, but got a different HTTP code", url)))
-		}
-		assertRequestFailure := func(url string) {
-			execInSleepPod(t, ns, buildGetRequestCmd(url),
-				assert.OutputContains(curlFailedMessage,
-					"Got a failure message as expected",
-					"Expect request to failed, but got a response"))
-		}
-		assertRequestTimeout := func(url string) {
-			execInSleepPod(t, ns, buildGetRequestCmd(url),
-				assert.OutputContains("504",
-					"Got expected 504 response since the request was timeout",
-					"Expect a timeout response with 504, but got a different one"))
-		}
-
 		ossm.DeployControlPlane(t)
 
-		t.LogStepf("Install sleep into %s", ns)
-		app.InstallAndWaitReady(t, app.Sleep(ns))
-
-		t.LogStep("Make request to www.redhat.com from sleep")
-		assertRequestSuccess("https://www.redhat.com/en")
+		t.LogStepf("Install sleep into %s", ns.Bookinfo)
+		sleep := app.Sleep(ns.Bookinfo)
+		app.InstallAndWaitReady(t, sleep)
+
+		t.LogStepf("Install httpbin in %s", ns.MeshExternal)
+		httpbin := app.HttpbinNoSidecar(ns.MeshExternal)
+		app.InstallAndWaitReady(t, httpbin)
+
+		t.LogStep("Make request to external httpbin from sleep")
+		httpbinHeadersUrl := fmt.Sprintf("http://%s.%s:8000/headers", httpbin.Name(), httpbin.Namespace())
+		assertRequestSuccess(t, sleep, httpbinHeadersUrl)
+
+		t.LogStep("Make sure that external httpbin was not discovered by Istio - it would happen if mesh-external namespaces was added to the SMMR")
+		retry.UntilSuccess(t, func(t test.TestHelper) {
+			shell.Execute(t,
+				fmt.Sprintf("istioctl pc endpoint deploy/sleep -n %s", ns.Bookinfo),
+				assert.OutputDoesNotContain(
+					fmt.Sprintf("%s.%s.svc.cluster.local", httpbin.Name(), httpbin.Namespace()),
+					"Httpbin was not discovered",
+					"Expected Httpbin to not be discovered, but it was."))
+		})
 
 		t.LogStepf("Patch outbound traffic policy to registry only")
```

Comment:
Can we add here a t.Log with this link, to be more understandable what does this: https://istio.io/latest/docs/tasks/traffic-management/egress/egress-control/#envoy-passthrough-to-external-services

Reply:
Done